tests/components/azure_event_hub/test_init.py

```diff
@@ -90,128 +38,173 @@ async def test_full_config(hass):
+            },
+        }
+    }
+    assert await async_setup_component(hass, DOMAIN, config)
 
-async def test_allowlist(hass, mock_batch, mock_call_later):
-    """Test an allowlist only config."""
-    process_queue = await _setup(
-        hass,
-        mock_call_later,
-        {
-            "include_domains": ["light"],
-            "include_entity_globs": ["sensor.included_*"],
-            "include_entities": ["binary_sensor.included"],
-        },
-    )
 
-    tests = [
-        FilterTest("climate.excluded", False),
-        FilterTest("light.included", True),
-        FilterTest("sensor.excluded_test", False),
-        FilterTest("sensor.included_test", True),
-        FilterTest("binary_sensor.included", True),
-        FilterTest("binary_sensor.excluded", False),
-    ]
+async def test_unload_entry(hass, entry, mock_create_batch):
+    """Test being able to unload an entry.
 
-    await _run_filter_tests(hass, tests, process_queue, mock_batch)
+    Queue should be empty, so adding events to the batch should not be called, this verifies that the unload, calls async_stop, which calls async_send and shuts down the hub.
+    """
+    assert await hass.config_entries.async_unload(entry.entry_id)
+    mock_create_batch.add.assert_not_called()
+    assert entry.state == ConfigEntryState.NOT_LOADED
 
 
-async def test_denylist(hass, mock_batch, mock_call_later):
-    """Test a denylist only config."""
-    process_queue = await _setup(
-        hass,
-        mock_call_later,
-        {
-            "exclude_domains": ["climate"],
-            "exclude_entity_globs": ["sensor.excluded_*"],
-            "exclude_entities": ["binary_sensor.excluded"],
-        },
+async def test_failed_test_connection(hass, mock_get_eventhub_properties):
+    """Test being able to unload an entry."""
+    entry = MockConfigEntry(
+        domain=azure_event_hub.DOMAIN,
+        data=SAS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
     )
-
-    tests = [
-        FilterTest("climate.excluded", False),
-        FilterTest("light.included", True),
-        FilterTest("sensor.excluded_test", False),
-        FilterTest("sensor.included_test", True),
-        FilterTest("binary_sensor.included", True),
-        FilterTest("binary_sensor.excluded", False),
-    ]
-
-    await _run_filter_tests(hass, tests, process_queue, mock_batch)
+    entry.add_to_hass(hass)
+    mock_get_eventhub_properties.side_effect = EventHubError("Test")
+    await hass.config_entries.async_setup(entry.entry_id)
+    assert entry.state == ConfigEntryState.SETUP_RETRY
 
 
-async def test_filtered_allowlist(hass, mock_batch, mock_call_later):
-    """Test an allowlist config with a filtering denylist."""
-    process_queue = await _setup(
+async def test_send_batch_error(hass, entry_with_one_event, mock_send_batch):
+    """Test a error in send_batch."""
+    mock_send_batch.reset_mock()
+    mock_send_batch.side_effect = EventHubError("Test")
```

Comment:
Maybe set None as the second side effect and handle another event to test that the code is able to continue and handle this exception without blowing up.